### PR TITLE
Improve logging

### PR DESCRIPTION
**Pull request by santib**
BEFORE:
image

AFTER:
image

PROS:
- In staging/production envs we get 1 log per line, instead of many

CONS:
- We lose details about views/partials rendering. Shouldn't be a problem in prod since we should use a monitoring tool for this purpose.

### CODE REVIEW

File: config/initializers/lograge.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  config.lograge.custom_payload do |controller|
+    request = controller.request
+    {
+      current_user_id: controller.try(:current_user).try(:id),
+      current_admin_user_id: controller.try(:current_admin_user).try(:id),
+      host: request.host,
+      path: request.path,
+      query_string: request.query_string,
+      referer: request.referer,
+      user_agent: request.user_agent
```
Comment (santib) [inline config/initializers/lograge.rb:14]: do we want this in our logs? or what information do we want?

File: config/initializers/lograge.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  config.lograge.enabled = true
+  config.lograge.custom_payload do |controller|
+    request = controller.request
+    {
+      current_user_id: controller.try(:current_user).try(:id),
+      host: request.host,
+      path: request.path,
+      query_string: request.query_string,
+      referer: request.referer,
+      user_agent: request.user_agent
```
Comment (santib) [inline config/initializers/lograge.rb:13]: do we want this in our logs? or what information do we want?

File: Gemfile [hunk 3]
```diff
@@ -19,6 +19,7 @@ gem 'flipper', '~> 0.28.3'
 gem 'flipper-active_record', '~> 0.28.0'
 gem 'flipper-ui', '~> 0.28.0'
 gem 'jbuilder', '~> 2.10'
+gem 'lograge', '~> 0.13.0'
```
Comment (rodrieiz) [inline Gemfile:22]: Hadn't you recommended us to specify the version with 2 digits and allow minor updates?

File: config/initializers/lograge.rb (hunk 2 above)
Comment (rodrieiz) [reply config/initializers/lograge.rb:13]: Is possible to add the time info?

**Review by rodrieiz — COMMENTED** [review]
Just a question, Do we need to remove code from the previous 'prints logs'?

File: config/initializers/lograge.rb (hunk 2 above)
Comment (santib) [reply config/initializers/lograge.rb:13]: good question. In the lograge docs it says
>It doesn't include any timestamp by default, instead it assumes you use a proper log formatter instead.

And this is what I see for example in AWS Cloudwatch:
image

File: Gemfile (hunk 3 above)
Comment (santib) [reply Gemfile:22]: good point. From https://semver.org/ it says:
>MINOR version when you add functionality in a backward compatible manner

so it should be safe to do that yes. I'll change it

**Review by rodrieiz — APPROVED** [review]

File: config/initializers/lograge.rb (hunk 2 above)
Comment (JulianPasquale) [reply config/initializers/lograge.rb:13]: Few questions here:
- Can que add the IP address? Might be useful to detect bots attacks or similar
- Would query_string include all the request params? I mean, from url and from a form/post request body?

**Review by JulianPasquale — APPROVED** [review]

File: config/initializers/lograge.rb (hunk 2 above)
Comment (santib) [reply config/initializers/lograge.rb:13]: - IIRC IP addresses are considered to be PII which goes against GDPR. I'd like this base app to be GDPR-compliant by default
- query_string only have the query params in the URL

**Review by enzofab91 — APPROVED** [review]

**Pull request merged**